test/runtests.jl

```diff
@@ -49,28 +42,44 @@ end
             return
         end
         ThreadingUtilities.pause()
+        @assert (counter += 1) < 1_000_000_000
     end
 end
 
-@testset "Internals" begin
-    @test ThreadingUtilities._atomic_store!(pointer(UInt[]), (), 1) == 1
-    @test ThreadingUtilities.ThreadTask() isa ThreadingUtilities.ThreadTask
-end
-
 @testset "ThreadingUtilities.jl" begin
     @time Aqua.test_all(ThreadingUtilities)
 
+    @testset "THREADPOOL" begin
+        @test isconst(ThreadingUtilities, :THREADPOOL) # test that ThreadingUtilities.THREADPOOL is a constant
+        @test ThreadingUtilities.THREADPOOL isa Tuple
+        @test eltype(ThreadingUtilities.THREADPOOL) === ThreadingUtilities.ThreadTask
+        @test length(ThreadingUtilities.THREADPOOL) == Sys.CPU_THREADS-1
+        for i = 1:length(ThreadingUtilities.THREADPOOL)
+            ThreadingUtilities.THREADPOOL[i] isa ThreadingUtilities.ThreadTask
+        end
+        if length(ThreadingUtilities.TASKS) > 0
+            @test @inferred(ThreadingUtilities.taskpointer(1)) isa Ptr{UInt}
+        end
+    end
+    
+    @testset "Internals" begin
+        @test ThreadingUtilities._atomic_store!(pointer(UInt64[]), (), 1) == 1
```

Comment:
@chriselrod You need to change the UInt64 to UInt so that tests will pass with 32-bit Julia binaries.

Reply:
Done